docs/cns.md

```diff
@@ -0,0 +1,110 @@
+# Chain Name Service Draft
+
+The goal of this module is to be a simple registry with the ability to transfer ownership. Auctions and other mechanisms are out of scope and can be implemented alongside or on top of this system (perhaps, in a custodial way).
+
+# State
+
+## Chain name
+
+Chain name contains information about the chain, most importantly, chain ID used to uniquly identify the chain and a list of peer nodes.
+
+Key: `chain-[chainID] -> Chain Name` 
+
+```go
+type ChainName struct {
+	ChainID string
+	Owner sdk.AccAddress
+	Peers []peer
+	SourceURL string
+	SourceCommitHash string
+}
+```
+
+## Offer
+
+Offers are used to transfer ownership of a chain name.
+
+```go
+type Offer struct {
+	ID int32
+	From sdk.AccAddress
+	Amount sdk.Coins
+	Status status // pending | rejected | approved
+}
+```
+
+# Messages
+
+## MsgEditChainName
+
+Chain owner can make edits to the chain name.
+
+```go
+type MsgEditChainName struct {
+	
+}
+```
+
+## MsgTransferChainName
+
+To transfer a name from one owner from another without creating an offer with a zero amount.
+
+```go
+type MsgTransferChainName struct {
+	From string
+	To string
+	ChainID string
+}
+```
+
+## MsgCreateOffer
+
+```go
+type MsgCreateOffer struct {
+	ChainID string
+	Amount sdk.Coins
+}
+```
+
+## MsgRejectOffer
+
+```go
+type MsgRejectOffer struct {
+	ChainID string
+}
+```
+
+## MsgApproveOffer
+
+```go
+type MsgApproveOffer struct {
+	ChainID string
+}
+```
+
+## MsgRemoveOffer
+
+```go
+type MsgRejectOffer struct {
+	ChainID string
+}
+```
+
+## MsgRegisterName
+
+To accuire a name that doesn't have an owner. By default each name has a price. When registering a name, amount equivalent to the price goes to the community pool.
```

Comment:
ENS uses an oracle now to ensure price stability. Until that feature was available they just locked ETH and gave it back once the name was revoked. Maybe that's a way that would be simple for this to begin? It would be unfortunate if early users were punished by paying in ATOM instead of the fiat equivalent at the time. On the other hand going to the community pool is pretty easy to justify.

Reply:
What data does the oracle bring in? Also I'm in favor of community pool.

Reply:
I think an oracle might be valuable in other areas more than here (confirming seednodes, potentially RPC endpoints) but in this case it was a USD/Crypto oracle so that prices of new names weren't volatile with the underlying crypto used to claim them.